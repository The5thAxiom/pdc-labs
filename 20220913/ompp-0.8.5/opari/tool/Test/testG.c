#include <omp.h>
#include <stdio.h>

int main() {
  int me, glob;

  #pragma omp parallel private(me,glob)
  {
    me = omp_get_thread_num();
    printf("hello from %d\n", me);

    #pragma omp single copyprivate(glob)
    {
      glob = me;
      printf("single1 from %d\n", me);
    }

    printf("hello from %d\n", me);

    #pragma omp single 
    {
      printf("single2 from %d\n", me);
    }

    #pragma omp single nowait
    {
      printf("single3 from %d\n", me);
    }

  }
}
