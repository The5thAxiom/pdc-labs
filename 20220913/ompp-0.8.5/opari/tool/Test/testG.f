       program testG
         integer me, glob
         integer omp_get_thread_num

         !$omp parallel private(me,glob)
           me = omp_get_thread_num()
           write(*,*) "hello from", me

           !$omp single
             glob = me
             write(*,*) "single1 from", me
           !$omp end single copyprivate(glob)

           write(*,*) "hello from", me

           !$omp single 
             write(*,*) "single2 from", me
           !$omp end single 

           !$omp single
             write(*,*) "single3 from", me
           !$omp end single nowait

         !$omp end parallel
       end program testG
