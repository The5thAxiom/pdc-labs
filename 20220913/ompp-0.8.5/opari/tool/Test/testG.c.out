#ifdef _POMP
#  undef _POMP
#endif
#define _POMP 200110

#include "testG.c.opari.inc"
#line 1 "testG.c"
#include <stdio.h>

int main() {
  int me, glob;

POMP_Parallel_fork(&omp_rd_50);
#line 7 "testG.c"
  #pragma omp parallel private(me,glob) POMP_DLIST_00050
{ POMP_Parallel_begin(&omp_rd_50);
#line 8 "testG.c"
  {
    me = omp_get_thread_num();
    printf("hello from %d\n", me);

POMP_Single_enter(&omp_rd_51);
#line 12 "testG.c"
    #pragma omp single copyprivate(glob)
{ POMP_Single_begin(&omp_rd_51);
#line 13 "testG.c"
    {
      glob = me;
      printf("single1 from %d\n", me);
    }
POMP_Single_end(&omp_rd_51); }
POMP_Barrier_enter(&omp_rd_51);
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_51);
POMP_Single_exit(&omp_rd_51);
#line 17 "testG.c"

    printf("hello from %d\n", me);

POMP_Single_enter(&omp_rd_52);
#line 20 "testG.c"
    #pragma omp single  nowait
{ POMP_Single_begin(&omp_rd_52);
#line 21 "testG.c"
    {
      printf("single2 from %d\n", me);
    }
POMP_Single_end(&omp_rd_52); }
POMP_Barrier_enter(&omp_rd_52);
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_52);
POMP_Single_exit(&omp_rd_52);
#line 24 "testG.c"

POMP_Single_enter(&omp_rd_53);
#line 25 "testG.c"
    #pragma omp single nowait
{ POMP_Single_begin(&omp_rd_53);
#line 26 "testG.c"
    {
      printf("single3 from %d\n", me);
    }
POMP_Single_end(&omp_rd_53); }
POMP_Single_exit(&omp_rd_53);
#line 29 "testG.c"

  }
POMP_Barrier_enter(&omp_rd_50);
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_50);
POMP_Parallel_end(&omp_rd_50); }
POMP_Parallel_join(&omp_rd_50);
#line 31 "testG.c"
}
